### PR TITLE
Improve Material selection controls

### PR DESCRIPTION
**Pull request by abarth**
- These controls now have proper radial reactions.
- You can drag the switch.
- The radio button animates properly.
- There's a demo in the Material Gallery

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie This patch makes the selection controls pretty good.  One issue I ran into is that for Switch, the tap gesture and the drag gesture are fighting each other, which means the drag is a little "sticky" when you start dragging it.  The way the UX is designed, there isn't any ambiguity in what to draw, but the gesture recognizers don't know that.

**Comment by abarth** [comment]
@HansMuller

File: examples/material_gallery/lib/demo/selection_controls_demo.dart [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    });
+  }
+
+  Widget build(BuildContext context) {
+    return new Column([
+      new Row([
+        new Checkbox(value: _checkboxValue, onChanged: _setCheckboxValue),
+        new Checkbox(value: false), // Disabled
+      ], justifyContent: FlexJustifyContent.spaceAround),
+      new Row([0, 1, 2].map((int i) {
```
Comment (Hixie) [inline examples/material_gallery/lib/demo/selection_controls_demo.dart]: `<int>[0, 1, 2]`

File: examples/material_gallery/lib/demo/selection_controls_demo.dart [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  }
+
+  void _setSwitchValue(bool value) {
+    setState(() {
+      _switchValue = value;
+    });
+  }
+
+  Widget build(BuildContext context) {
+    return new Column([
```
Comment (Hixie) [inline examples/material_gallery/lib/demo/selection_controls_demo.dart]: `<Widget>[...]`

File: examples/material_gallery/lib/demo/selection_controls_demo.dart [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+      new Row([
```
Comment (Hixie) [inline examples/material_gallery/lib/demo/selection_controls_demo.dart]: `<Widget>[...]`

File: examples/material_gallery/lib/demo/selection_controls_demo.dart [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        new Checkbox(value: false), // Disabled
+      ], justifyContent: FlexJustifyContent.spaceAround),
+      new Row([0, 1, 2].map((int i) {
+        return new Radio<int>(
+          value: i,
+          groupValue: _radioValue,
+          onChanged: _setRadioValue
+        );
+      }).toList(), justifyContent: FlexJustifyContent.spaceAround),
+      new Row([0, 1].map((int i) {
```
Comment (Hixie) [inline examples/material_gallery/lib/demo/selection_controls_demo.dart]: `<int>[0, 1]`

**Comment by Hixie** [comment]
LGTM, though I was flagging a bit near the end of that patch.

Should these get converted to Material-layer ink splashes, or are they like slider splashes?

**Comment by abarth** [comment]
> Should these get converted to Material-layer ink splashes, or are they like slider splashes?

They're like the slider splashes.  The one for the switch moves with the thumb of the switch and is above the track.  The one for the radio button starts with the radius of the button.  We could move the checkbox one, but it seems better to have them be consistent.  I'm going to see if I can share some splash code between RenderToggleable and RenderSlider.

**Comment by Hixie** [comment]
sounds good

**Pull request merged**